test/e2e/subscription_e2e_test.go

```diff
@@ -1410,14 +1410,31 @@ var _ = Describe("Subscription", func() {
 		_, err := fetchCatalogSourceOnStatus(crClient, catsrc.GetName(), testNamespace, catalogSourceRegistryPodSynced)
 		require.NoError(GinkgoT(), err)
 
+		duplicateCatalogSources := 10
+		var (
+			wg                  sync.WaitGroup
+			cleanupDuplicatesFn []cleanupFunc
+		)
+		wg.Add(duplicateCatalogSources)
+
 		// Create duplicates of the CatalogSource
-		for i := 0; i < 10; i++ {
-			duplicateCatsrc, _, duplicateCatSrcCleanup := newCatalogSourceWithDependencies(GinkgoT(), kubeClient, crClient, "podconfig", testNamespace, permissions)
-			defer duplicateCatSrcCleanup()
+		for i := 0; i < duplicateCatalogSources; i++ {
```

Comment:
It's not immediately clear to me why 10 was chosen so I wanted to hold off on simply decrementing the counter without realizing the full context. I think a simpler solution to cut down on this long running e2e test is to simply decrement the number of duplicate CatalogSource resources that are created.